### PR TITLE
Use label to mark WorkloadDefinition in workload

### PR DESCRIPTION
**Pull request by zzxwill**
Use label "workload.oam.dev/type" to makr the type
of WorkloadDefinition for a Workload, and replace
the corresponding annotation

### CODE REVIEW

File: api/types/types.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ const (
 	AnnAPIVersion = "definition.oam.dev/apiVersion"
 	AnnKind       = "definition.oam.dev/kind"
 
-	// Indicate which workloadDefinition generate from
-	AnnWorkloadDef = "workload.oam.dev/name"
 	// Indicate which traitDefinition generate from
 	AnnTraitDef = "trait.oam.dev/name"
+	// WorkloadTypeLabel indicates the type of the workloadDefinition
+	WorkloadTypeLabel = "workload.oam.dev/type"
```
Comment (wonderflow) [inline api/types/types.go:20]: 确认一下，vela 里这里的WorkloadType 跟 OAM的WorkloadType 是不是同一个东西。

vela里存在一个shortname的情况，当前定义在cuelang template里面了。

**Comment by resouer** [comment]
workload/trait definition 的名字作为 type，short name 这个可以 deprecate 了。

**Comment by wonderflow** [comment]
synced with @ryanzhang-oss , and he will fix this as @resouer  said

**Comment by wonderflow** [comment]
Closing

1. we have [`workload.oam.dev/type` in oam-runtime](https://github.com/crossplane/oam-kubernetes-runtime/blob/master/pkg/oam/labels.go#L32) to indicate the workload type.
2. we will deprecate the short name #418

**Pull request closed without merging**